src/components/ShareButton.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <div class="share-button-container">
+    <button class="action-btn -share" @click="handleShare">
+      <img src="/icons/share.svg" alt="Share" />
+    </button>
+    <div
+      v-show="showCopyAlert"
+      class="copy-alert"
+      :class="{ show: copyAlertVisible }"
+    >
+      Link copied to clipboard!
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+/**
+ * Reusable share button that uses the native Web Share API when available,
+ * with clipboard fallback. Shows a fade-in alert when link is copied.
+ */
+@Component
+export default class ShareButton extends Vue {
+  @Prop({ type: String, default: '' })
+  title!: string;
+
+  @Prop({ type: String, default: '' })
+  text!: string;
+
+  @Prop({ type: String, default: '' })
+  url!: string;
+
+  showCopyAlert = false;
+  copyAlertVisible = false;
+
+  // Animation timing constants
+  private readonly AlertFadeDurationMs = 300;
+  private readonly AlertDisplayDurationMs = 2500;
+
+  handleShare(): void {
+    const shareUrl =
+      this.url || (typeof window !== 'undefined' ? window.location.href : '');
+
+    if (typeof window !== 'undefined' && window.navigator?.share) {
+      // Use the Web Share API - a native browser API that allows websites to share content
+      // using the device's native sharing mechanisms (e.g., share sheet on mobile devices)
+      // Supported on most modern mobile browsers and some desktop browsers
+      window.navigator
+        .share({
```

Comment:
I hadn't heard of this API before - it doesn't do anything on desktop, but it's very cool on mobile!